### ISSUE TITLE
Gradient accumulation

### ISSUE THREAD

**Issue by yrrIsHere** [ZiqiaoPeng/DualTalk#9]
Hi, 

Thanks for your excellent work! 

I noticed something that might be wrong when training with gradient accumulation, between line 59 and 64 in `main.py`, where your code is below:

```python
optimizer.zero_grad()
loss.backward()
loss_log.append(loss.item())
if i % args.gradient_accumulation_steps == 0:
    optimizer.step()
    optimizer.zero_grad()
```

This might be a wrong way to use gradient accumulation, as u use `zero_grad()` each time before `backward()`. I think a better way is below:

```python
if i % args.gradient_accumulation_steps == 0:
    optimizer.zero_grad()

loss = loss / args.gradient_accumulation_steps
loss.backward()
loss_log.append(loss.item())

if (i + 1) % args.gradient_accumulation_steps == 0 or (i + 1 == len(train_loader)):
    optimizer.step()
```

Thanks for your patience.
